File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/billing/shared/bill-invoice-return.model.ts

import * as moment from 'moment';
import { BillInvoiceReturnItemsModel } from './bill-invoice-return-items.model';
import { BillingTransactionItem } from './billing-transaction-item.model';

export class BillInvoiceReturnModel {
  public BillReturnId: number = 0;
  public CreditNoteNumber: number = 0;
  public RefInvoiceNum: number = 0;
  public PatientId: number = null;
  public FiscalYearId: number = null;
  public FiscalYear: string = null;
  public BillingTransactionId: number = null;
  public SubTotal: number = 0;
  public DiscountAmount: number = 0;
  public TaxableAmount: number = 0;
  public TaxTotal: number = 0;
  public TotalAmount: number = 0;
  public Remarks: string = null;
  public CounterId: number = null;
  public CreatedBy: number = null;
  public CreatedOn: string = null;
  public IsActive: boolean = true;
  public IsRemoteSynced: boolean = false;
  public TaxId: number = null;
  public InvoiceCode: string = null;
  public Tender: number = null;
  public ReturnedItems: Array<BillingTransactionItem> = [];//Rajesh:7Aug19 -- this can be removed after removing partial return
  public ReturnInvoiceItems: Array<BillInvoiceReturnItemsModel> = [];//Sud-1May'21 for Credit Note

  public PaymentMode: string = null;
  public IsInsuranceBilling: boolean = false;
  public InsuranceProviderId: number = null;
  public BillStatus: string = null;
  public DiscountReturnAmount: number = 0;
  public DiscountFromSettlement: number = 0;
  public ReturnCashAmount: number = 0;
  public ReturnCreditAmount: number = 0;
  public PriceCategoryId: number = 0;
  public TransactionType: string = "";
  public IsCoPayment: boolean = false;

  public SchemeId: number = 0;//Sud:22March'23--For new billingStructure.
  public OrganizationId: number = null;
  public ClaimCode: number = null;
  constructor() {
    this.CreatedOn = moment().format("YYYY-MM-DD HH:mm:ss");
  }
}

File: Code/Components/DanpheEMR.ServerModel/BillingModels/BillInvoiceReturnModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

/*
 File: BillInvoiceReturnModel.cs
 Created: 5May'18 <sudarshan>
 Description: Created after IRD and HAMS Update
 Remarks: Name is similar as that of BillingReturnRequestModel, change that later on.
     */


namespace DanpheEMR.ServerModel
{
    public class BillInvoiceReturnModel
    {
        [Key]
        public int BillReturnId { get; set; }
        public int? CreditNoteNumber { get; set; }
        public int RefInvoiceNum { get; set; }
        public int? PatientId { get; set; }
        public string FiscalYear { get; set; }
        public int? FiscalYearId { get; set; }
        public int? BillingTransactionId { get; set; }
        public double? SubTotal { get; set; }
        public double? DiscountAmount { get; set; }
        public double? TaxableAmount { get; set; }
        public double? TaxTotal { get; set; }
        public double? TotalAmount { get; set; }
        public string Remarks { get; set; }
        public int? CounterId { get; set; }
        public int? CreatedBy { get; set; }
        public DateTime? CreatedOn { get; set; }
        public bool? IsActive { get; set; }
        [NotMapped]
        public float Tender { get; set; }

        [NotMapped]
        public List<BillingTransactionItemModel> ReturnedItems { get; set; }
        public PatientModel Patient { get; set; }//added for IRD-sud:6May'18

        public int? TaxId { get; set; }
        public string InvoiceCode { get; set; }

        public bool? IsRemoteSynced { get; set; }
        public bool? IsRealtime { get; set; }//sud: 10May'18

        //public bool? IsLocalSynced { get; set; }//sud: 10May'18

        [NotMapped]
        public int? PartialReturnTxnId { get; set; }
    }
}
